swarm/fuse/swarmfs_unix.go

```diff
@@ -72,88 +73,97 @@ func NewMountInfo(mhash, mpoint string, sapi *api.Api) *MountInfo {
 	return newMountInfo
 }
 
-func (self *SwarmFS) Mount(mhash, mountpoint string) (*MountInfo, error) {
-
+func (swarmfs *SwarmFS) Mount(mhash, mountpoint string) (*MountInfo, error) {
+	log.Info(fmt.Sprintf("swarmfs: mounting hash %s at mount point %s", mhash, mountpoint))
 	if mountpoint == "" {
 		return nil, errEmptyMountPoint
 	}
 	cleanedMountPoint, err := filepath.Abs(filepath.Clean(mountpoint))
 	if err != nil {
 		return nil, err
 	}
+	log.Trace(fmt.Sprintf("swarmfs mount: cleanedMountPoint %s", cleanedMountPoint))
 
-	self.swarmFsLock.Lock()
-	defer self.swarmFsLock.Unlock()
+	swarmfs.swarmFsLock.Lock()
+	defer swarmfs.swarmFsLock.Unlock()
 
-	noOfActiveMounts := len(self.activeMounts)
+	noOfActiveMounts := len(swarmfs.activeMounts)
+	log.Debug(fmt.Sprintf("swarmfs mount: # active mounts %d", noOfActiveMounts))
 	if noOfActiveMounts >= maxFuseMounts {
 		return nil, errMaxMountCount
 	}
 
-	if _, ok := self.activeMounts[cleanedMountPoint]; ok {
+	if _, ok := swarmfs.activeMounts[cleanedMountPoint]; ok {
 		return nil, errAlreadyMounted
 	}
 
-	log.Info(fmt.Sprintf("Attempting to mount %s ", cleanedMountPoint))
-	_, manifestEntryMap, err := self.swarmApi.BuildDirectoryTree(mhash, true)
+	log.Debug(fmt.Sprintf("swarmfs mount: getting manifest tree"))
+	_, manifestEntryMap, err := swarmfs.swarmApi.BuildDirectoryTree(mhash, true)
 	if err != nil {
 		return nil, err
 	}
 
-	mi := NewMountInfo(mhash, cleanedMountPoint, self.swarmApi)
+	log.Debug(fmt.Sprintf("swarmfs mount: building mount info"))
+	mi := NewMountInfo(mhash, cleanedMountPoint, swarmfs.swarmApi)
 
 	dirTree := map[string]*SwarmDir{}
 	rootDir := NewSwarmDir("/", mi)
-	dirTree["/"] = rootDir
+	log.Trace(fmt.Sprintf("swarmfs mount: rootDir %v", rootDir))
```

Comment:
so `dirTree["/"] = rootDir` was the bug?

Reply:
correct, this one caused the bug that showed `/` as a directory entry